esrally/client/asynchronous.py

```diff
@@ -328,9 +337,13 @@ async def perform_request(
         # Converts all parts of a Accept/Content-Type headers
         # from application/X -> application/vnd.elasticsearch+X
         # see https://github.com/elastic/elasticsearch/issues/51816
-        if self.distribution_version is not None and self.distribution_version >= versions.Version.from_string("8.0.0"):
-            _mimetype_header_to_compat("Accept", request_headers)
-            _mimetype_header_to_compat("Content-Type", request_headers)
+        # Not applicable to serverless
+        if not self.is_serverless:
+            if self.distribution_version is not None and (
```

Comment:
Can we assume that `distribution_version` will be defined now?

Reply:
I think we can for the async client, just because we currently only create them in one place where we already know the distribution flavor/version. I mainly did this to be consistent between the two, but I think a better approach is to just check whether the value is a valid identifier as I've done in https://github.com/elastic/rally/pull/1738/commits/5f2db54f79ae9dc1b5c2bf3882319268a0945c65